pysoa/server/action/status.py

```diff
@@ -147,6 +169,44 @@ def run(self, request):
 
         return status
 
+    def _check_client_settings(self, request):
+        """
+        This method checks any client settings configured for this service to call other services, calls the `status`
+        action of each configured service with `verbose: False` (which guarantees no further recursive status checking),
+        adds that diagnostic information, and reports any problems. To include this check in your status action, define
+        `check_client_settings = BaseStatusAction._check_client_settings` in your status action class definition.
+        """
+        if not request.client.settings:
+            # There's no need to even add diagnostic details if no client settings are configured
+            return
+
+        self.diagnostics['services'] = {}
```

Comment:
Nitpick, I called the top-level key that I nested pysoa service diagnostics `pysoa_services`. I don't feel strongly about the name, but Eventbrite ops will need to adjust if we change it.

I have a slight preference for the more specific `pysoa_services` vs. `services` to support the use case where pysoa is used together with other service frameworks.

Reply:
I have used `pysoa_services` in the health checks for legacy SOA services (such as Decision Service), to intentionally indicate that they are non-legacy-SOA services. However, as this is a built-in specifically for PySOA services only (and as it's part of our public OSS software), I think `services` fits better here.

Reply:
👍